### PR TITLE
Adds validation of description values.

### LINKED ISSUE #424: Validation: only one of value, parallelValue, groupedValue, and structuredValue should be present
On the same level an object should have either a value or a structuredValue, not both.

### PR DESCRIPTION
**Pull request by justinlittman**
closes #424

**NOTE:  Changes to openapi.yml require updating openapi.yml for sdr-api and dor-services-app and generating models - see README.**

## Why was this change made? 🤔
Per Arcadia.

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

Unit

### CODE REVIEW

**Comment by justinlittman** [comment]
This is going to have to be held until after migration for testing against prod.

**Comment by justinlittman** [comment]
Tested against QA, stage, and prod with no issues.

File: spec/cocina/models/validators/description_values_validator_spec.rb [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+    it 'does not raise' do
+      validate
+    end
+  end
+
+  describe 'when an invalid Description' do
+    # let(:clazz) { Cocina::Models::RequestDRO }
+    let(:props) { invalid_desc_props }
+    # let(:dro_props) { { description: invalid_desc_props } }
```
Comment (mjgiarlo) [inline spec/cocina/models/validators/description_values_validator_spec.rb:100]: Are the commented-out variables cruft?

File: lib/cocina/models/validators/description_values_validator.rb [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+          end
+        end
+
+        def validate_obj(obj, path)
+          validate_hash(obj, path) if obj.is_a?(Hash)
+          validate_array(obj, path) if obj.is_a?(Array)
+        end
+
+        def validate_values(hash, path)
+          return unless hash.to_a.filter { |key, value| %i[value groupedValue structuredValue parallelValue].include?(key) && value.present? }.size > 1
```
Comment (mjgiarlo) [inline lib/cocina/models/validators/description_values_validator.rb:56]: How would you feel about using `Hash#count` for this? So, like:

```suggestion
  return if hash.count { |key, value| %i[value groupedValue structuredValue parallelValue].include?(key) && value.present? } < 2
```

File: lib/cocina/models/validators/description_values_validator.rb [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+          error_paths << path_to_s(path).to_s
```
Comment (mjgiarlo) [inline lib/cocina/models/validators/description_values_validator.rb:58]: Do you need the `#to_s` on the end? I assumed based on the method name that `#path_to_s` might return a string.

**Comment by justinlittman** [comment]
Thanks @mjgiarlo. All changes made.

**Review by mjgiarlo — APPROVED** [review]

**Pull request merged**